hal/src/sercom/v2/uart/reg.rs

```diff
@@ -0,0 +1,411 @@
+//! Register-level access to UART configuration
+
+use super::{BaudMode, BitOrder, CharSizeEnum, Parity, StopBits};
+
+use crate::sercom::v2::*;
+use crate::target_device as pac;
+
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+use pac::sercom0::usart::ctrla::MODE_A;
+
+#[cfg(feature = "min-samd51g")]
+use pac::sercom0::usart_int::ctrla::MODE_A;
+
+use crate::time::Hertz;
+
+pub struct Registers<S: Sercom> {
```

Comment:
Actually, it looks like you made the type `pub`, but the module is private. Again, I think consistency is the important thing. Either approach should be fine.